### PR TITLE
PDA customisation options + Set Pose big box

### PR DESCRIPTION
**Pull request by Mwahahahaha**
Ports over 5 PDA icons from Polaris, BS12, and Eris.
These have been added in as character setup options which change the object icon of your character's PDA.
Underneath said option, people can now also choose whether their PDAs start with their ringers and news ringers on or off, and can customise their ringtones from there.

The menu on the upper right may look a little more cluttered though, what with 5 more options being stuffed in there.

image

<details>
<summary>Default</summary>

</details>
<details>
<summary>Slate</summary>

</details>
<details>
<summary>Slim</summary>

</details>
<details>
<summary>Old</summary>

</details>
<details>
<summary>Rugged</summary>

</details>
<details>
<summary>Smart</summary>

</details>
<details>
<summary>Glass</summary>

</details>

Also turns the Set Pose text box into a nice big message box to see your glorified poses. Inputted text is retained instead of left blank, but the input proc automatically highlights all of it anyway if you need to quickly change it.

<details>
<summary>Set Pose box</summary>
<br>
<br>
<br>No, this is not what this is about. If you're someone like CERTAIN people on the server, then your pose can end up looking like this:
<br>
<br>

which is pretty d*rn tooting difficult to oversee in a single-line text box.
</details>

Note:
- Clicking save character seems to screw up the HTML-escaped characters for the custom ringtones like the ' apostrophe, but this is using the legacy (not SQL) save system

Todo:
- Separate PDA tab in the setup to minimise clutter
- Dealing with PDA cartridge icons

### CODE REVIEW

**Comment by Mwahahahaha** [comment]
!review

**Comment by alsoandanswer** [comment]
Maybe put all PDA customizations all under one tab like

PDA : 
Type: smart
Ringer: whatev

And so on.

**Comment by Mwahahahaha** [comment]
> Maybe put all PDA customizations all under one tab like
> 
> **PDA** :
> Type: smart
> Ringer: whatev
> 
> And so on.

I suppose that's one thing to consider. Would it be a separate section under General, a separate tab along with General and Loadout and bla bla bla?

> And if it were up to me, as a spriter, I would personally have rugged for the default.
> Although a PDA sprite overhaul would be for another time.

_aw man   but I always liked the default one we have now hrm hm m rmpph php_
It could use some beautifying up though. Looks a little like something just past the Exadv1 days. Or like, the tg ERP update of 2013.
So by that I mean just some more shading would probably help it. I've already actually started doing that by adding some slight colouration on the edges of the emitted screen.

I'll see what I can do with the cartridges.

**Comment by alsoandanswer** [comment]
Yeah it would be a seperate tab among general and stuff

**Comment by alsoandanswer** [comment]
It's good practice to seperate the set-pose update from the PDA one and put it in two seperate PRs. They're practically unrelated. Makes things easier to review and merge.

**Comment by Mwahahahaha** [comment]
I know. ;(
I was actually going to do that because I know that it's not a good idea to make hybrid abomination PRs for sake logistics and documentation,



but then I like 


ran out of disk space. so

**Comment by alsoandanswer** [comment]
Wh-holy shit

Well...did you try clearing your recycle bin and other stuff?

I once had like 3.2 GB taken up by a torrent of CIV VI and I finally got around to deleting it.

Or, an alternative would to be just making a .txt folder and just copypasting what you did and then once the PRs done you can just slap it back inside

Maybe get a thumbdrive? Only suggestions though. I have no idea what you or your computer is dealing with.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -74,6 +75,22 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		if (pen)
 			to_chat(user, "There is \a [pen] in the pen slot.")
 
+/obj/item/device/pda/New(var/mob/living/carbon/human/H)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/PDA/PDA.dm]: this should be on initialize

**Comment by skull132** [comment]
Not a fan of PDA icon options. The PDA is a distinct piece of equipment issued to all NT employees, much like the ID is. It should _not_ be something customizable, for that comes at the cost of losing out on the corporate image. (Or at the very least, the icons should not be such a mishmash of differing styles.)

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (Mwahahahaha) [reply code/game/objects/items/devices/PDA/PDA.dm]: Yoch, I might be missing something to make it work for Initialize() besides adding an argument for the ../human

**Comment by Mwahahahaha** [comment]
I don't know how to make it so that it doesn't show up as a garbled mess on the character setup. I don't know how records and pAI information does this. I tried adding html_decode to the raw variable that is applied to the saved pref.variable but that just causes problems by adding &amp or whatever it is on the setup every time the setup is refreshed.

**Comment by Mwahahahaha** [comment]
> Not a fan of PDA icon options. The PDA is a distinct piece of equipment issued to all NT employees, much like the ID is. It should _not_ be something customizable, for that comes at the cost of losing out on the corporate image. (Or at the very least, the icons should not be such a mishmash of differing styles.)

See I considered this. But I thought about it for a little while. Forgive me first for I am no business student, but...

Given the nature of the corporation's various employees from many different backgrounds, many different biological structures, and many different ways of thinking — not just with humans alone but across each of the species — I would've figured that it's quite important to accommodate these different things; not every end-user is going to work in the same way and it is actually beneficial to be able to cover those corners when it comes to accessibility and user convenience, ESPECIALLY in a working environment, let alone a working environment with already many different needs and criteria just by occupations alone. Those working in EVA, for example, may arguably benefit a little more with buttons on their PDA rather than a touchscreen.

There are many different schools of approach to this, however, but the one I wanted to propose was one in which a system was designed to allow for as many users as possible. After all, in order to maintain a big megacorporation status, you probably may need to still cover a lot of grounds for your many different 'customers', monopoly or not. There's a reason why luxury brands or otherwise those that are catered for a very specific demographic aren't often that big.

That also goes without saying in regards to corporate image that we've already resorted to using contractors with their own identities to (partially) run the station along with different uniform options for each occupation and different brands of prostheses, some of which don't even have that particular requirement, so I'm not too sure. In any case, you can see examples of companies providing many different models of products with very different designs (e.g., shoes, vehicles, handheld devices) for very different needs. Wouldn't you say that it's much more impressive for a company to be able to come up with these things than if they imposed a single model on everyone, especially when they're supposed to be a company that takes in a wide variety of employees and perhaps even caters to just as wide a variety of customers on the business side of things? Don't you think that would be a beneficial corporate image to hold? We've also already established in the lore that we have these cosmopolitan locations as the most immediate locations within the vicinity: I can guarantee you that NO-ONE will take a monopoly/oligarchy unless it satisfies their needs. Not everyone is some materialistic teenager who simply jumps up on the hype of the next model to replace their old one. That dynamic could certainly still exist, but it isn't the only one. This PR doesn't have to change that aspect.
And of course, there's also the matter of visitors and other non-crew elements to remember.

**Comment by alsoandanswer** [comment]
I can probably whip up a resprite which just matches all the artstyles together and this should end up fine.

Also I have to probably change RedSec PDAs to BlueSec...

**Comment by skull132** [comment]
> I can probably whip up a resprite which just matches all the artstyles together and this should end up fine.
> 
> Also I have to probably change RedSec PDAs to BlueSec...

This is the preferred option.

I don't really have the time to read your reply in full, but a quick diagonal reveals that it's not really touching on what I touched. These are different PDAs with wholly different styles from different stations. They do not work out well with our stuff. Purely from an OOC art perspective, this is a bad idea.

Further, the other customization options largely fit in with some general aesthetic, if they need to pursue it. Various items need to pursue it to various degrees. It is my opinion that PDAs should definitely be relatively uniform in appearance. Real life examples are irrelevant (and up for debate regardless).

**Comment by Arrow768** [comment]
Something that could be interesting is to be able to indirectly customize what PDA you get via the faction selection.
i.e. Employees from the same faction always have the same PDA

**Comment by NonQueueingMatt** [comment]
@Mwahahahaha Are you continuing this?

**Comment by Mwahahahaha** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
Last commits on 5 october + conflicts. Reopen when this is being worked on.

**Pull request closed without merging**